internal/pkg/agent/cmd/inspect.go

```diff
@@ -299,9 +299,51 @@ func getProgramsFromConfig(log *logger.Logger, agentInfo *info.AgentInfo, cfg *c
 		return nil, err
 	}
 	composableWaiter.Wait()
+
+	// add the fleet-server input to default programs list
+	// this does not correspond to the actual config that fleet-server uses as it's in fleet.yml and not part of the assembled config (cfg)
+	fleetCFG, err := cfg.ToMapStr()
+	if err != nil {
+		return nil, err
+	}
+	if fleetInput := getFleetInput(fleetCFG); fleetInput != nil {
+		ast, err := transpiler.NewAST(fleetInput)
+		if err != nil {
+			return nil, err
+		}
+		router.programs["default"] = append(router.programs["default"], program.Program{
+			Spec: program.Spec{
+				Name: "fleet-server",
+				Cmd:  "fleet-server",
+			},
+			Config: ast,
+		})
+	}
+
```

Comment:
I see, fleet-server is a special case :/
We might want to revisit the whole diagnostic with the v2 and where can improve, it was an afterthought that make it harder to extend.

Reply:
yeah, i've brought some of the points up with @blakerouse, but we need to work in how we handle this as well